methods.rb

```diff
@@ -0,0 +1,116 @@
+module Enumerable
+  def my_each
+    i = 0
+    length.times do
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
```

Comment:
- [ ] `my_each` is not working as the original `each` method, also is not passing all test cases.

### `each` is an Enumerator that has these characteristics: 
- Returns an Enumerator if no block is given.
```
[1, 2, 3].each 
# #<Enumerator: [1, 2, 3]:each>
````
- It does not change or modify the original array/range

- it returns the array/range/hash itself if a block is given.
```
[1, 2, 3].each { |num| num }
# [1, 2, 3]
(1..3).each { |num| num * 2 }
# 1..3
````
- it applies the block for each element inside the array/range/hash
```
[1, 2, 3].each { |num| puts num }
# 1
# 2
# 3
# => [1, 2, 3]
```
[Enumerator#each](https://ruby-doc.org/core-2.7.2/Enumerator.html#method-i-each)